x-pack/plugins/enterprise_search/server/lib/indices/utils/get_index_data.ts

```diff
@@ -57,7 +62,9 @@ export const getIndexData = async (
   const indicesNames = returnHiddenIndices
     ? Object.keys(totalIndices)
     : Object.keys(totalIndices).filter(
-        (indexName) => !(totalIndices[indexName]?.settings?.index?.hidden === 'true')
+        (indexName) =>
+          !(totalIndices[indexName]?.settings?.index?.hidden === 'true') ||
+          (alwaysShowPattern?.index_pattern && indexName.includes(alwaysShowPattern.index_pattern))
```

Comment:
I believe this override pattern should be at the start of the index name not in-between.
ie. 
- `.ent-search-documents-123123` should shown
- `random-prefix.ent-search-documents-123123` shouldn't be shown. 

Instead of `indexName.includes` you can change it with `indexName.startsWith`

Reply:
Yes. I missed to notice this. Will change. 

I believe it may probably require to change the test cases as well.